timeline-chart/src/components/time-graph-annotation.ts

```diff
@@ -0,0 +1,144 @@
+import * as PIXI from "pixi.js-legacy"
+import { TimeGraphComponent, TimeGraphElementPosition, TimeGraphComponentOptions } from "./time-graph-component";
+import { TimeGraphRow } from "./time-graph-row"
+
+export interface TimeGraphAnnotationComponentOptions extends TimeGraphComponentOptions {
+    position: TimeGraphElementPosition
+}
+
+export interface TimeGraphAnnotationStyle extends TimeGraphComponentOptions {
+    symbol?: string
+    size?: number
+    color?: number
+}
+
+/*
+ * This is only implementing a subset of the tick elements so far
+ */
+export class TimeGraphAnnotationComponent extends TimeGraphComponent {
+
+    // TODO: make a map of the display objects
+    // e.g. cross-14-black etc... 
+    // and then use this._displayObject.addChild(cross-14-black) in the draw.
+    // if performance is an issue
+
+    protected _size: number;
+
+    constructor(id: string,
+        protected _options: TimeGraphAnnotationComponentOptions,
+        protected _style: TimeGraphAnnotationStyle = { color: 0, size: 14, symbol: 'cross' },
+        protected _row: TimeGraphRow,
+        displayObject?: PIXI.Graphics) {
+        super(id, displayObject);
+        this._size = _style.size || 14;
+        // called to ensure consistency. Only the X component is used from options.
+        this.update(_options);
+    }
+
+    update(opts: TimeGraphAnnotationComponentOptions, style?: TimeGraphAnnotationStyle): void {
+        if (opts) {
+            this._options.position.x = opts.position.x;
+            this._options.position.y = this._row.position.y + (this._row.height * 0.5);
+        }
+        if (style) {
+            this._style.color = style.color;
+            this._style.size = style.size;
+            this._style.symbol = style.symbol;
+        }
+        super.update();
+    }
+
+    render(): void {
+        const { symbol } = this._style as TimeGraphAnnotationStyle;
+        const size = this._size;
+        const x = this._options.position.x - (size * 0.5);
+        const y = this._options.position.y - (size * 0.5);
+        if (symbol == 'circle') {
```

Comment:
In the other libraries that we use, the symbol size is the radius, not the circumference. Would it be better to do the same?

Also, I think it would be more intuitive if the (x, y) given to the draw methods is the center point, not the top-left corner.